FogCreek/Glitch-Community
# classy-afrovenator: Add billing tab to /settings

## Links
* [classy-afrovenator.glitch.me](https://classy-afrovenator.glitch.me)
* [[ch6209]](https://app.clubhouse.io/glitch/story/6209/add-billing-tab-to-settings): Add billing tab to /settings

## GIF/Screenshots:
**To come. The backend is returning `500`s for the `/glitchPro` endpoints. Will update when Justin is in and can investigate.**

## Changes:
This adds a "Subscription" tab to /settings behind the `pufferfish` feature flag.

## How To Test:
*Note that this is the dogfood implementation of this feature, so review should be focused on "Does it work well enough to test internally and iterate on?" rather than polish and cleverness.*

### Testing Subscription Tab Visibility
To see the tab:

1. Log in
2. Make sure you're in [@testing-team](https://glitch.com/@testing-team)
3. Make sure the `pufferfish` flag is enabled on [classy-afrovenator.glitch.me/secret](https://classy-afrovenator.glitch.me/secret)

| Case  | Expected Outcome | 
| ------------- | ------------- |
| In Testing Team + Enabled the feature on /secret  | Subscription tab visible |
| In Testing Team + Disabled the feature on /secret  | Subscription tab NOT visible |
| NOT In Testing Team + Enabled the feature on /secret  | Subscription NOT tab visible  |
| NOT In Testing Team + Disabled the feature on /secret  | Subscription NOT tab visible |

### Testing Creating + Cancelling a Subscription
**To come. The backend is returning `500`s for the `/glitchPro` endpoints. Will update when Justin is in and can investigate.**

## Feedback I'm looking for:
- Can you create a subscription with test data?
- Can you cancel that subscription?
- Do the `useStripe` and `useSubscription` hooks feel ok to you?
